### ISSUE TITLE
osslsigncode 2.7-1 occurs segmentation fault.

### ISSUE THREAD

**Issue by bitamind** [msys2/MINGW-packages#19285]
### Description / Steps to reproduce the issue

osslsigncode 2.7-1 occurs segmentation fault.
I guess it is fixed on osslsigncode2.8dev commit [osslsigncode@cdb7557](https://github.com/mtrojnar/osslsigncode/commit/cdb75578e9a5decf9031a03d0c59a9f23465ccdd).
I tested [this](https://github.com/mtrojnar/osslsigncode/actions/runs/6434796969) mingw binary, it is fixed.
could you port this?
thank you.

### Expected behavior

```
$ ./osslsigncode sign -verbose -pass ${pin_sign} \
 -pkcs11engine ${APPDIR}/osslsigncode/pkcs11.dll \
 -pkcs11module "${PROGRAMFILES}/OpenSC Project/PKCS11-Spy/pkcs11-spy.dll" \
 -certs ${APPDIR}/osslsigncode/cert/sign.pem -key 1:2 -h sha1 -t http://timestamp.digicert.com \
 -in Portable-VirtualBox.exe -out Portable-VirtualBox_signed.exe
Unable to load provider: legacy
Warning: Legacy mode disabled
Engine "pkcs11" set.
Enter PKCS#11 key PIN for Digital Signature Key:
Connecting to http://timestamp.digicert.com
Succeeded
```

### Actual behavior

```
$ pacman -Ss osslsigncode
ucrt64/mingw-w64-ucrt-x86_64-osslsigncode 2.7-1 [installed]
    Tool for Authenticode signing of PE, CAB and MSI files (mingw-w64)

$ osslsigncode sign -verbose ${pin_sign} \
 -pkcs11engine ${APPDIR}/osslsigncode/pkcs11.dll \
 -pkcs11module "${PROGRAMFILES}/OpenSC Project/PKCS11-Spy/pkcs11-spy.dll" \
 -certs ${APPDIR}/osslsigncode/cert/sign.pem -key 1:2 -h sha1 -t http://timestamp.digicert.com \
 -in Portable-VirtualBox.exe -out Portable-VirtualBox_signed.exe
Engine "pkcs11" set.
Segmentation fault
```

### Verification

- [X] I have verified that my MSYS2 is up-to-date before submitting the report (see https://www.msys2.org/docs/updating/)

### Windows Version

MINGW64_NT-10.0-19045

### MINGW environments affected

- [ ] MINGW64
- [ ] MINGW32
- [x] UCRT64
- [ ] CLANG64
- [ ] CLANG32
- [ ] CLANGARM64

**Comment by Biswa96** [comment]
Hi @bitamind, Thank you for reporting the issue. I have added a pull request to fix this issue. Would you like to test the packages from GitHub Actions artifacts of that pull request? Link https://github.com/msys2/MINGW-packages/actions/runs/7153284198

**Comment by bitamind** [comment]
I've tested the ucrt binary,  but it is not fixed.
```
$ osslsigncode_fixucrt sign -verbose -pass ${pin_sign} \
 -pkcs11engine ${APPDIR}/osslsigncode/pkcs11.dll \
 -pkcs11module "${PROGRAMFILES}/OpenSC Project/OpenSC/pkcs11/opensc-pkcs11.dll" \
 -certs ${APPDIR}/osslsigncode/certs/sign.pem -key 1:2 -h sha1 \
 -t http://timestamp.digicert.com \
 -in Portable-VirtualBox.exe -out Portable-VirtualBox_signed.exe
Engine "pkcs11" set.
Segmentation fault
```
so I made pkcs12 key, and tested.
```
$ osslsigncode sign -verbose -pass ${pin_sign} \
 -pkcs12 ${APPDIR}/osslsigncode/certs/cscert.p12 \
 -t http://timestamp.digicert.com \
 -in Portable-VirtualBox.exe -out Portable-VirtualBox_signed.exe
Connecting to http://timestamp.digicert.com
Succeeded

$ osslsigncode_fixucrt sign -verbose -pass ${pin_sign} \
 -pkcs12 ${APPDIR}/osslsigncode/certs/cscert.p12 \
 -t http://timestamp.digicert.com \
 -in Portable-VirtualBox.exe -out Portable-VirtualBox_signed.exe
Connecting to http://timestamp.digicert.com
Succeeded
```
I am facing to different probrem.
I was mistaken because the problem was resolved in 2.8dev.
I'm sorry for giving the wrong report.
thank you.

**Comment by Biswa96** [comment]
No problem, the package will be updated when upstream make a new release.
